receiver/hostmetricsreceiver/internal/scraper.go

```diff
@@ -75,7 +43,7 @@ type ResourceScraperFactory interface {
 
 	// CreateMetricsScraper creates a resource scraper based on this
 	// config. If the config is not valid, error will be returned instead.
-	CreateMetricsScraper(ctx context.Context, logger *zap.Logger, cfg Config) (ResourceScraper, error)
+	CreateMetricsScraper(ctx context.Context, logger *zap.Logger, cfg Config) (receiverhelper.ResourceMetricsScraper, error)
```

Comment:
nit: Consider renaming this method to `CreateResourceMetricsScraper` for clarity?